src/plugins/discover/public/application/components/top_nav/get_top_nav_links.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { i18n } from '@kbn/i18n';
+import { showOpenSearchPanel } from './show_open_search_panel';
+import { getSharingData } from '../../helpers/get_sharing_data';
+import { unhashUrl } from '../../../../../kibana_utils/public';
+import { DiscoverServices } from '../../../build_services';
+import { Adapters } from '../../../../../inspector/common/adapters';
+import { SavedSearch } from '../../../saved_searches';
+import { onSaveSearch } from './on_save_search';
+import { GetStateReturn } from '../../angular/discover_state';
+import { IndexPattern } from '../../../kibana_services';
+
+/**
+ * Helper function to build the top nav links
+ */
+export const getTopNavLinks = ({
+  getFieldCounts,
+  indexPattern,
+  inspectorAdapters,
+  navigateTo,
+  savedSearch,
+  services,
+  state,
+}: {
+  getFieldCounts: () => Promise<Record<string, number>>;
```

Comment:
I'm struggling to understand where `getFieldCounts` is defined? Can you point me?

Reply:
yes, it the legacy stuff I didn't dare to touch:

https://github.com/elastic/kibana/blob/3e83e59953a96673278b7abb311affaa4bd03e20/src/plugins/discover/public/application/angular/discover.js#L340

I also think that we might remove this with runtime field changes, so I decided to leave it for now

Reply:
Ah, Github collapsed that file entirely, so I missed it :( Thanks 👍